targets/TARGET_ONSEMI/TARGET_NCS36510/gpio_irq_api.c

```diff
@@ -156,11 +159,12 @@ int gpio_irq_init(gpio_irq_t *obj, PinName pin, gpio_irq_handler handler, uint32
      * If mbed layer wants to call this function repeatedly for setting multiple DIO lines as input
      * then change this setting to  obj->GPIOMEMBASE->W_IN |= obj->pinMask. All parameter setting needs to change from = to |=
      */
-    obj->GPIOMEMBASE->W_IN = obj->pinMask;
-    obj->GPIOMEMBASE->IRQ_ENABLE_SET = obj->pinMask;
-    obj->GPIOMEMBASE->IRQ_EDGE = obj->pinMask;
-    obj->GPIOMEMBASE->IRQ_POLARITY_SET = (obj->pinMask);
-    obj->GPIOMEMBASE->ANYEDGE_SET = IO_NONE;
+
+    obj->GPIOMEMBASE->W_IN |= obj->pinMask;
```

Comment:
Assuming the datasheet I have is correct this is wrong, as is the comment about - you should never be performing read-modify-write on any of these registers.